src/txn/txn_rollback_to_stable.c

```diff
@@ -804,32 +775,18 @@ __rollback_abort_newer_updates(
     }
 
     /*
-     * If we have a ref with no page, or the page is clean, find out whether the page has any
-     * modifications that are newer than the given timestamp. As eviction writes the newest version
-     * to page, even a clean page may also contain modifications that need rollback. Such pages are
-     * read back into memory and processed like other modified pages.
+     * If we have a ref with clean page, find out whether the page has any modifications that are
+     * newer than the given timestamp. As eviction writes the newest version to page, even a clean
+     * page may also contain modifications that need rollback.
      */
-    if ((page = ref->page) == NULL || !__wt_page_is_modified(page)) {
-        if (!__rollback_page_needs_abort(session, ref, rollback_timestamp)) {
-            __wt_verbose(session, WT_VERB_RTS, "%p: page skipped", (void *)ref);
-#ifdef HAVE_DIAGNOSTIC
-            if (ref->page == NULL && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY)) {
-                WT_RET(__wt_page_in(session, ref, 0));
-                if (ref->page->type == WT_PAGE_ROW_LEAF)
-                    __rollback_verify_ondisk_page(session, ref->page, rollback_timestamp);
-                WT_TRET_BUSY_OK(__wt_page_release_evict(session, ref, WT_READ_NO_SPLIT));
-            }
-#endif
-            return (0);
-        }
-
-        /* Page needs rollback, read it into cache. */
-        if (page == NULL) {
-            WT_RET(__wt_page_in(session, ref, 0));
-            local_read = true;
-        }
-        page = ref->page;
+    WT_ASSERT(session, ref->page != NULL);
+    page = ref->page;
+    if (!__wt_page_is_modified(page) &&
+      !__rollback_page_needs_abort(session, ref, rollback_timestamp)) {
+        __wt_verbose(session, WT_VERB_RTS, "%p: page skipped", (void *)ref);
```

Comment:
Don't we need to increment the stat representing pages skipped ?

Reply:
I think this is OK, I think the "skipped pages" stat is only intended for pages the tree-walk itself skips.

Reply:
Yes, the current stat is to represent how many pages are skipped as part of tree walk, so no need to increment here.